feature: build coach mark

jumolis opened the pull request:
I think it is mostly ready.  Please let me know what else is needed.

jumolis commented:
I included it because I thought I saw it in the console inviteTooltip.js, but you are right.  It is much cleaner without the extra element.  I made the changes and I will check them in shortly.

jumolis commented:
I think it is mostly ready.  Please let me know what else is needed.

aarmour commented on demos/src/demo.scss:
```diff
@@ -2,3 +2,30 @@ $o-assets-global-path: '/bower_components/';
+  font-family: proxima_nova, 'Helvetica Neue', Helvetica, Arial, sans-serif;
+}
+
+#cm-bottom {
```
id selectors produce a warning by the linter. It won't fail the build, but it would be nice to remove the warnings. Can this be refactored to use class selectors?

aarmour commented on src/scss/_mixins.scss:
```diff
@@ -1,13 +1,78 @@
+			}
+		}
+
+		&__bottom::before {
```
`bottom`, `top`, `left`, and `right` are modifiers, so they should be `--bottom`, etc.

aarmour commented on src/scss/_mixins.scss:
```diff
@@ -1,13 +1,78 @@
+
+			.close_icon {
+				float: right;
+			  margin-top: -3px;
```
Whitespace

aarmour commented on src/scss/_mixins.scss:
```diff
@@ -1,13 +1,78 @@
+			}
+
+			.title {
+    		padding-bottom: 7px;
```
Whitespace

aarmour commented:
I didn't mark all of the lines with mixed tabs/spaces. Please review and fix.

aarmour commented:
We're now using [o-component-handler](https://github.com/pearson-higher-ed/o-component-handler) to make registration and DOM upgrade of components easier. Refer to this [example](https://github.com/Pearson-Higher-Ed/o-component-handler/blob/master/examples/ExampleComponent.js) for registering a component. Let me know if you have questions about how it works.

jumolis commented:
Still need to add o-component-handler, but I think everything else is corrected.

jumolis commented:
@aarmour I think that completes everything on the list.  Let me know if you need anything else.

aarmour commented:
There's still a minor alignment issue:
<img width="275" alt="screen shot 2015-11-18 at 3 41 43 pm" src="https://cloud.githubusercontent.com/assets/1048508/11256760/102b1e12-8e0b-11e5-8cf5-a8128deaa3c1.png">

Also, I noticed that the positions seem to change unexpectedly depending on the viewport width, which can also be observed when the page is zoomed or refreshed in some cases.
<img width="1553" alt="screen shot 2015-11-18 at 3 42 24 pm" src="https://cloud.githubusercontent.com/assets/1048508/11256761/103eed2a-8e0b-11e5-86bb-ac229f22220e.png">

jumolis commented:
@aarmour I think I fixed the alignment issue.  Also, the issue you were seeing about things moving on the page I think was caused by scroll, not necessarily viewport width.  I fixed that too.
